### PR TITLE
Tests and fixes for end-user context functionality.

### PR DESCRIPTION
**Pull request by DonQueso89**
Test coverage for end-user context switching. This revealed some flaws in the functionality. 
Specifically: 

* the `dbroles.py` module was not taking into account the case in which there is already a transaction pending (which is the case during test runs). A ROLLBACK then reverted (because postgres does not support nested transactions and a ROLLBACK will just rollback all pending transactions) a lot of state (mostly declared cursors), which the app was expecting to be present.
* the active user was preserved on the database connection object between requests, causing subsequent requests not to switch the database user 
* When a session has no user email, we did not switch to an ANONYMOUS_ROLE

### CODE REVIEW

File: src/rest_framework_dso/views.py [hunk 1]
```diff
@@ -328,7 +328,7 @@ def _parse_accept_crs(self, value: Optional[str]) -> Optional[crs.CRS]:
 
         try:
             accept_crs = crs.CRS.from_string(value)
-        except ExternalValueError as e:
+        except (ExternalValueError, SyntaxError) as e:
```
Comment (DonQueso89) [inline src/rest_framework_dso/views.py:331]: This is to fix a regression caused by a change in the errors that are  raised by django-gisserver

Comment (lbam) [reply src/rest_framework_dso/views.py:331]: Looks like you have the wrong version of django-gisserver installed.

File: src/dso_api/dbroles.py [hunk 2]
```diff
@@ -92,40 +90,59 @@ def activate_end_user(cls, user_connection: BaseDatabaseWrapper, *, log_source:
...
+                    logger.debug("Starting transaction for user-context: %s", role_name)
+                    c.execute("BEGIN TRANSACTION;")
+                else:
+                    # If we dont create a transaction here, we need to
+                    # remember which role was active before the context
+                    # started
+                    c.execute("SELECT current_user;")
+                    original_role = c.fetchone()[0]
+                    logger.debug(
+                        "Transaction already in progress storing original role: %s", original_role
```
Comment (lbam) [inline src/dso_api/dbroles.py:134]:
```suggestion
                        "Transaction already in progress. Storing original role: %s", original_role
```

File: src/dso_api/middleware.py [hunk 3]
```diff
@@ -24,7 +24,6 @@ def __call__(self, request: HttpRequest):
             # get_token_scopes is a data attribute, not a method.
             scopes = request.get_token_scopes or []
             request.user_scopes = UserScopes(request.GET, scopes, self._all_profiles)
-
```
Comment (lbam) [inline src/dso_api/middleware.py:27]: Unnecessary change, please revert to keep the patch smaller.

File: src/requirements_dev.txt [hunk 4]
```diff
@@ -1,30 +1,25 @@
...
+# To update, run:
 #
-#    pip-compile --generate-hashes --output-file=requirements_dev.txt requirements_dev.in
+#    pip-compile --allow-unsafe --generate-hashes --output-file=requirements_dev.txt requirements_dev.in
 #
-amsterdam-schema-tools[django]==5.6.2 \
-    --hash=sha256:8c3830baa77103e4274b6bfd53ed098944187771339ddfa9036c72cb9b2345cc \
-    --hash=sha256:f2e16ef8ce0b00829ddbf3f94512ff0a8a46f795c0c9dd0ce8cbb7ff0c9fe9b0
-    # via -r ./requirements.txt
+amsterdam-schema-tools[django]==5.5.2 \
```
Comment (lbam) [inline src/requirements_dev.txt:7]: Are we moving back to 5.5.2?

Comment (DonQueso89) [reply src/requirements_dev.txt]: whops, fixed

**Comment by lbam** [comment]
When I run the tests locally, I get 5 errors:

```
FAILED src/tests/test_user_context.py::test_anonymous_when_no_token - django.db.utils.ProgrammingError: role "scope_openbaar" already exists
FAILED src/tests/test_user_context.py::test_user_when_token - django.db.utils.ProgrammingError: role "scope_openbaar" already exists
FAILED src/tests/test_user_context.py::test_user_switches_on_consecutive_requests - django.db.utils.ProgrammingError: role "scope_openbaar" already exists
FAILED src/tests/test_user_context.py::test_internal_user_when_subject_without_role - django.db.utils.ProgrammingError: role "scope_openbaar" already exists
FAILED src/tests/test_user_context.py::test_permission_error_for_external_unknown_users - django.db.utils.ProgrammingError: role "scope_openbaar" already exists
```

File: src/requirements_dev.in [hunk 5]
```diff
@@ -1,17 +1,9 @@
...
-# Debugging
-django-debug-toolbar == 3.7.0
-django-extensions == 3.2.1
+black
+isort
+pytest-sugar==0.9.6
+termcolor
+pre-commit 
+django-debug-toolbar
+django-extensions
```
Comment (lbam) [inline src/requirements_dev.in:9]: I think there should be versions here. Otherwise, every `make requirements` may produce a different `requirements_dev.txt` and black/isort may also change their rules.

Comment (DonQueso89) [reply src/requirements_dev.in]: done

File: src/dso_api/middleware.py (hunk 3 above)
Comment (DonQueso89) [reply src/dso_api/middleware.py:27]: done

**Review by Yrob — APPROVED** [review]
